plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx

```diff
@@ -111,6 +113,31 @@ export const Stepper = (stepperProps: StepperProps) => {
   const [errors, setErrors] = useState<undefined | FormValidation>();
   const styles = useStyles();
 
+  const stringifiedSecrets = JSON.stringify(secrets);
+
+  // Because secrets can be defined in the schema, we need to make sure that they
+  // are included in the validation process. So we merge the secrets and the formData
+  // together in validation.
+  const customValidator = useMemo<ValidatorType>(
+    () => ({
+      isValid: (schema, formData, rootSchema) =>
+        validator.isValid(schema, { ...formData, ...secrets }, rootSchema),
+      rawValidation: (schema, formData) =>
+        validator.rawValidation(schema, { ...formData, ...secrets }),
+      validateFormData: (formData, schema, customFormats, transformErrors) =>
+        validator.validateFormData(
+          { ...formData, ...secrets },
+          schema,
+          customFormats,
+          transformErrors,
+        ),
+      // @deprecated
+      toErrorList: validator.toErrorList,
```

Comment:
just note that this doesn't preserve `this` === `validator` - maybe not a problem in practice

Reply:
Yeah it's not called at all, it's just there to make typescript happy.